catalog/db_branch_scanner.go

```diff
@@ -48,28 +48,29 @@ func NewDBBranchScanner(tx db.Tx, branchID int64, commitID CommitID, opts *DBSca
 }
 
 func getRelevantCommitsCondition(tx db.Tx, branchID int64, commitID CommitID) (string, error) {
-	var maxCommitId CommitID
+	var branchMaxCommitID CommitID
 	var commits []string
 	var commitsWhere string
 	if commitID == UncommittedID || commitID == CommittedID {
-		maxCommitId = MaxCommitID
+		branchMaxCommitID = MaxCommitID
 	} else {
-		maxCommitId = commitID
+		branchMaxCommitID = commitID
 	}
-	sql := "SELECT commit_id::text FROM catalog_commits WHERE branch_id = $1 AND commit_id <= $2 ORDER BY commit_id"
-	err := tx.Select(&commits, sql, branchID, maxCommitId)
+	// commit_id name is changed so that sorting will be performed on the numeric value, not the string value (where "10" is less than "2")
+	sql := "SELECT commit_id::text as str_commit_id FROM catalog_commits WHERE branch_id = $1 AND commit_id <= $2 ORDER BY commit_id"
+	err := tx.Select(&commits, sql, branchID, branchMaxCommitID)
 	if err != nil {
 		panic(err)
 	}
 	if commitID == UncommittedID {
 		commits = append(commits, strconv.FormatInt(int64(MaxCommitID), 10))
 	}
-	if len(commits) > MaxCommitsInFilter {
+	if len(commits) < MaxCommitsInFilter {
```

Comment:
```suggestion
	if len(commits) <= MaxCommitsInFilter {
```
better matches my expectations of "max".